sdk/src/azure/iot/az_iot_common.c

```diff
@@ -119,7 +119,7 @@ AZ_NODISCARD az_result az_iot_message_properties_next(
   {
     *out_name = AZ_SPAN_EMPTY;
     *out_value = AZ_SPAN_EMPTY;
-    return AZ_ERROR_IOT_END_OF_PROPERTIES;
+    return (az_result)AZ_ERROR_IOT_END_OF_PROPERTIES;
```

Comment:
Is this cast required? If so, that is very unfortunate and requires discussion.

Reply:
Yes, with the current solution, the casts are required. I'll follow up and discuss.